src/backend/XmlHttpRequest.ts

```diff
@@ -136,6 +153,14 @@ export default class XmlHttpRequest extends BaseFileSystem implements FileSystem
     deprecationMessage(deprecateMsg, "XmlHttpRequest", { index: typeof(listingUrlOrObj) === "string" ? listingUrlOrObj : "file index as an object", baseUrl: prefixUrl});
 
     this._index = FileIndex.fromListing(listing);
+
+    this._requestFileAsync = (preferXHR || !('fetch' in global))
+      ? (p: string, type: string, cb: BFSCallback<any>) => asyncDownloadFile(this.getXhrPath(p), type, cb)
```

Comment:
`(p, type, cb) => asyncDownloadFile(this.getXhrPath(p), type, cb)` gets translated into the following JavaScript code:

```javascript
var _this = this;
function(p, type, cb) { asyncDownloadFile(_this.getXhrPath(p), type, cb); }
```

Since `_requestFileAsync` is always called on the `this` object, you can use an ordinary `function` and tell TypeScript the expected type of `this`:

```typescript
// Note: 'this' is a meta-argument
function(this: XmlHttpRequest, p: string, type: string, cb: BFSCallback<any>) { return asyncDownloadFile(this.getXhrPath(p), type, cb); }
```

JavaScript:

```javascript
function(p, type, cb) { return asyncDownloadFile(this.getXhrPath(p), type, cb); }
```

This code avoids an unneeded variable capture in the closure, and is a bit cleaner in the compiled code.

You don't have to do this for the PR to be accepted; I'll do it post-merge if it's annoying.